File: acwing/蓝桥杯每日一题2025/week1/6118.蛋糕游戏/6118.蛋糕游戏.py
N = 5 *int (1e5) + 10
s = [0] * N

t = int(input())

for _ in range(t):
    n = int(input())
    a = [0] + list(map(int, input().split()))
    res = float("inf")
    l = n // 2 + 1
    for i in range(1, n + 1):
        s[i] = s[i - 1] + a[i]
        if i >= l:
            res = min(res, s[i] - s[i - l])

    print(res, s[n] - s[0] - res)

File: acwing/蓝桥杯每日一题2025/week1/6118.蛋糕游戏/6118.蛋糕游戏.cpp
#include <algorithm>
#include <iostream>
#include <cstring>

using namespace std;

const int N = 5 * 1e5 + 10;

typedef long long LL;

LL s[N];

int main()
{
    int t;
    cin >> t;
    while (t--)
    {
        int n;
        cin >> n;
        LL res = 1e15;
        int l = n / 2 + 1;
        for (int i = 1; i <= n; i++)
        {
            int x;
            cin >> x;
            s[i] = s[i - 1] + x;
            if (i >= l)
                res = min(res, s[i] - s[i - l]);
        }
        cout << res << ' ' << s[n] - res << endl;
    }

    return 0;
}